core/src/main/java/io/questdb/cairo/wal/ApplyWal2TableJob.java

```diff
@@ -558,10 +579,17 @@ private long processWalSql(TableWriter tableWriter, WalEventCursor.SqlInfo sqlIn
                 try {
                     switch (cmdType) {
                         case CMD_ALTER_TABLE:
-                            operationExecutor.executeAlter(tableWriter, sql, seqTxn);
+                            final String matViewInvalidationReason = operationExecutor.executeAlter(tableWriter, sql, seqTxn);
+                            if (matViewInvalidationReason != null) {
+                                mvRefreshTask.operation = MatViewRefreshTask.INVALIDATE;
+                                mvRefreshTask.invalidationReason = matViewInvalidationReason;
+                            }
                             return -1;
                         case CMD_UPDATE_TABLE:
-                            return operationExecutor.executeUpdate(tableWriter, sql, seqTxn);
+                            final long rowsAffected = operationExecutor.executeUpdate(tableWriter, sql, seqTxn);
+                            mvRefreshTask.operation = MatViewRefreshTask.INVALIDATE;
```

Comment:
```suggestion
if (rowsAffected != 0) {
                            mvRefreshTask.operation = MatViewRefreshTask.INVALIDATE;
```

Reply:
Addressed in cfc9486372752e33d92034a5c841685b248ec1e7